### PR TITLE
docs(core): structure docs on creating libraries

### PR DESCRIPTION
**Pull request by isaacplmann**
_[Please make sure you have read the submission guidelines before posting an PR](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#submit-pr)_

> _Please make sure that your commit message follows our format._

> Example: `fix(nx): must begin with lowercase`

Adds documents about creating libraries

### CODE REVIEW

File: docs/shared/workspace/creating-libraries.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+### 2. Visualizing Architecture
+
+The `nx dep-graph` command generates a graph of how apps and libraries depend on each other. If most of your code lives in a few giant libraries, this visualization doesn't provide much value.
+
+### 3. Enforcing Constraints
+
+You can enforce constraints on how different types of libraries depend on each other [using tags](/{{framework}}/workspace/structure/monorepo-tags). Following pre-determined conventions on what kind of code can go in different types of libraries allows your tagging system to enforce good architectural patterns.
+
+Also, each library defines its own API, which allows for encapsulating logic that other parts of codebase can not access.
```
Comment (ZackDeRose) [inline docs/shared/workspace/creating-libraries.md:21]: Possibly related: ability to assign 'code ownership' of specific libs to a given team/developer

File: docs/shared/workspace/grouping-libraries.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+Libraries should be grouped by _scope_. A library's scope is either application to which it belongs or (for larger applications) a section within that application.
+
+## Example Workspace
+
+Let's use Nrwl Airlines as an example organization. This organization has two apps, `booking` and `check-in`. In the Nx workspace, libraries related to `booking` are grouped under a `libs/booking` folder, libraries related to `check-in` are grouped under a `libs/check-in` folder and libraries used in both applications are placed in `libs/shared`. You can also have nested grouping folders, (i.e. `libs/shared/seatmap`).
+
+The purpose of these folders is to help with organizing by scope. We recommend grouping libraries together which are (usually) updated together. It helps minimize the amount of time a developer spends navigating the folder tree to find the right file.
+
+```txt
```
Comment (ZackDeRose) [inline docs/shared/workspace/grouping-libraries.md:11]: Might be good to mention `@nrwl/angular:move` as a way of moving existing libs around

File: docs/shared/workspace/library-types.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
+# Library Types
+
+There are many different types of libraries in a workspace. In order to maintain a certain sense of order, we recommend having a small number of types, such as the below four (4) types of libraries:
```
Comment (ZackDeRose) [inline docs/shared/workspace/library-types.md:3]: Asset libs?

**Review by ZackDeRose — APPROVED** [review]
Left some thoughts, but in general LGTM!

File: docs/shared/workspace/grouping-libraries.md (hunk 2 above)
Comment (isaacplmann) [reply docs/shared/workspace/grouping-libraries.md:11]: Yes, good point.  I had meant to include that and forgot

File: docs/shared/workspace/library-types.md (hunk 3 above)
Comment (isaacplmann) [reply docs/shared/workspace/library-types.md]: I added a note giving explicit permission for people to come up with their own types.

**Review by isaacplmann — COMMENTED** [review]
This is weird UX.  I didn't intentionally start a review on my own PR...

**Pull request merged**